src/Nethermind/Nethermind.Optimism/CL/Rpc/OptimismOutputAtBlock.cs

```diff
@@ -0,0 +1,39 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using Nethermind.Core.Crypto;
+using Nethermind.Optimism.CL;
+
+namespace Nethermind.Optimism.Cl.Rpc;
+
+public sealed record OptimismOutputAtBlock
+{
+    public required byte[] Version { get; init; }
+    public required Hash256 OutputRoot { get; init; }
+    public required L2BlockRef BlockRef { get; init; }
+    public required Hash256 WithdrawalStorageRoot { get; init; }
+    public required Hash256 StateRoot { get; init; }
+    public required OptimismSyncStatus Status { get; init; }
+}
+
+public sealed record OptimismOutputV0
+{
+    public static readonly byte[] Version = new byte[32];
+
+    public required Hash256 StateRoot { get; init; }
+    public required Hash256 MessagePasserStorageRoot { get; init; }
+    public required Hash256 BlockHash { get; init; }
+
+    public Hash256 Root()
```

Comment:
Nice that it's a method. Shows that some execution might occur.
Have you considered `[SkipInit]`? 128 bytes. Maybe we don't need to clear them up?

Reply:
We can add it given that we don't need the buffer array to be initialized at all.